distributed/scheduler.py

```diff
@@ -5155,14 +5229,11 @@ def handle_long_running(
             else:
                 ts.prefix.duration_average = (old_duration + compute_duration) / 2
 
-        occ = ws.processing[ts]
-        ws.occupancy -= occ
-        self.total_occupancy -= occ
+        ws.remove_from_processing(ts)
```

Comment:
TODO: Double-check this change. Combined with the comment below it feels like it might cause issues.